repo/plugin.video.nzbdav/resources/lib/router.py

```
@@ -2162,13 +2162,23 @@ def _json_object(response):
+    probe = str(xml_text).lower()
+    return "<!doctype" in probe or "<!entity" in probe
+
+
 def _xml_root_name(response):
     """Return the unqualified root XML tag name, lowercased."""
     import xml.etree.ElementTree as ET  # nosec B405 - trusted service response
 
     try:
+        if _contains_xml_declaration_markup(response):
+            raise ValueError("DTD and entity declarations are not allowed")
         root = ET.fromstring(response)  # nosec B314 - trusted service response
```

Ruff (0.15.18)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)